### DISCUSSION TITLE
Фильтр списка с ключом value

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#292] [Как это сделать?] [answered]
Как отфильтровать данные с ключом value? Т.е. взять только те элементы списка, где value != 0

{

  "value": "4742810000",
  "transaction_id": 21212121

}

через listFilter(list: $#nfv; key: "value" != "0") почему-то не фильтрует.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Дело в том, что название элемента в rowList который нужно фильтрануть совпадает с управляющей командой в параметре `key`.

Управляющих 3 шт: `key`, `value` и `item`. 

`value` как управляющая команда означает оценивать содержимое каждой строки целиком — поэтому и не работает.

Что бы сказать, что у нас элемент называется `value` или `item` или `key` — нам нужно сказать, что это элемент и отдельным параметром передать как он называется:

```
= : listFilter(list: $#nfv; key: "item" != "0"; item: "value")
```
